### DISCUSSION TITLE
{{filecurrent}} rather than {{linkcurrent}}

### DISCUSSION THREAD

**Discussion by jdeml** [chhoumann/quickadd#360] [Q&A] [answered]
Is there a way to grab the current file name rather than a link to the current file?

The way that I'm attempting to use quickadd is as follows:

As I'm typing a note I'll have a section 
### Foo

I then want to create a note called Foo (or add to the existing Foo note) the text that falls under the Foo heading.

In my Capture Format I was expecting to be able to do something like:

![[{{pathcurrent}}#{{value}}]]
Which would give me something like ![[mypath/currentfile#Foo]]

but the only way I can see to grab the current file name seems to be {{linkcurrent}}.  This results in brackets that don't allow me to specify the header I hope to link to as it returns [[mypath/currentfile]] and I can't find a way to specify the header.  What I get is, 
!{{linkcurrent}}#Foo]] results in 
![[mypath/currentfile]]#Foo]]   (The brackets in before #Foo being the issue.)

Is there a better / different way to grab the path to the current file?

Thanks in advance.

**Comment by jdeml** [comment, marked as answer]
I found a solution to my own problem using a Templater function in the quickadd Capture Format.  The function was posted by stefantigro on reddit [(link to reddit post)](https://www.reddit.com/r/ObsidianMD/comments/z3hmhq/auto_parent_templater_template_that_can_be/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button):
```
<%* function getLastOpenFile(){ const lastActiveFile = app.workspace.lastActiveFile; 
// If the last active file is different from the current one, then it can be used 
if (lastActiveFile !== null && lastActiveFile.basename !== tp.file.title ) { const lastActiveFileBaseName = lastActiveFile.basename; console.log(`No match on last active file, using it!. (${lastActiveFileBaseName} !== ${tp.file.title})`); return lastActiveFileBaseName; } 
// If it's the same, then in the history the last opened one can be retrieved. 
const lastNameParts = app.workspace.lastOpenFiles[0].split("/"); 
console.log( "Last active filed matched, using:", lastNameParts); 
const name = lastNameParts[lastNameParts.length-1].replaceAll(".md", ""); return name; } -%>

![[<% getLastOpenFile() %>#{{value}}]]
```

**Reply by erauner12** [reply]
Thanks for this! Works great for me. Only question now is, how do take the name of that new zettelkasten note that was created (using this template) and add it directly to the spot of the current file that I created it from (effectively creating a two way link).

**Reply by jdeml** [reply]
erauner12, The following code in the original doc should create a link to the newly created doc. You may want to put this in a template and link the commands, but as I'm not using this way, I'll let you figure out what is best there.

`$=dv.pages().sort(p => p.file.ctime, 'desc').file.link.limit(1)`

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

QuickAdd has this natively now: `{{FILENAMECURRENT}}` gives you the raw name of the file you triggered the capture from - no brackets, so you can build the embed yourself. Your capture format becomes:

```markdown
![[{{FILENAMECURRENT}}#{{value}}]]
```

There is also `{{LINKSECTION}}`, which resolves to `[[Note#Heading]]` for the heading your cursor is currently under - so if you trigger the capture while sitting under `### Foo`, `!{{LINKSECTION}}` gets you the embed without typing the heading name at all.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#filenamecurrent

No Templater function needed anymore, so I'd use this instead of the workspace hack above. Closing as resolved.
